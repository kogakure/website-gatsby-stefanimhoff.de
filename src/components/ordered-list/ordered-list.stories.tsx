import React from 'react';

import { OrderedList } from '.';

export default {
  component: OrderedList,
  title: 'Lists/Ordered List',
};

export const Default = () => (
  <OrderedList>
    <li>
      Early Stoics
      <OrderedList>
        <li>Zeno of Citium</li>
        <li>Chrysippus</li>
        <li>Chrysippus</li>
      </OrderedList>
    </li>
  </OrderedList>
);
