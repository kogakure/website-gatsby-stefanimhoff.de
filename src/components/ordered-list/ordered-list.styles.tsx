import styled from 'styled-components';

import { fontSize, mediaQuery } from '../../layout/tokens';

const OrderedListContainer = styled.ol`
  font-size: ${fontSize[3]};
  padding-left: 1em;

  @media (${mediaQuery.tablet}) {
    padding-left: 0;
  }

  li & {
    padding-left: 2rem;
  }
`;

export const Styled = {
  OrderedListContainer,
};
