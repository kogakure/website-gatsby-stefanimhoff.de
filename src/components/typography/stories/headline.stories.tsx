import React from 'react';

import { Headline } from '..';

export default {
  title: 'Typography/Headline',
};

export const Default = () => <Headline>Headline</Headline>;

export const NoMargin = () => <Headline noMargin>Headline</Headline>;
