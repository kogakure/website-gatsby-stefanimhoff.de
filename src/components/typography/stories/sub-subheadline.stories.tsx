import React from 'react';

import { SubSubheadline } from '..';

export default {
  title: 'Typography/SubSubheadline',
};

export const Default = () => <SubSubheadline>SubSubheadline</SubSubheadline>;

export const NoMargin = () => (
  <SubSubheadline noMargin>SubSubheadline</SubSubheadline>
);
