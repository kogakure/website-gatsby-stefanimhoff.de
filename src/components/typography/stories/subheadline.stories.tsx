import React from 'react';

import { Subheadline } from '..';

export default {
  title: 'Typography/Subheadline',
};

export const Default = () => <Subheadline>Subheadline</Subheadline>;

export const NoMargin = () => <Subheadline noMargin>Subheadline</Subheadline>;
