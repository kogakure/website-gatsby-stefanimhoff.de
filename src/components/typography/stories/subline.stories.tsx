import React from 'react';

import { Subline } from '..';

export default {
  title: 'Typography/Subline',
};

export const Default = () => <Subline>Subline</Subline>;
