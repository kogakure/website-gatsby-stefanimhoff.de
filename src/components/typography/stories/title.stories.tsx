import React from 'react';

import { Title } from '..';

export default {
  title: 'Typography/Title',
};

export const Default = () => <Title>Title</Title>;

export const NoMargin = () => <Title noMargin>Title</Title>;
