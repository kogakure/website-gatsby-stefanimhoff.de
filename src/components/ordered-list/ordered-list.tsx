import * as React from 'react';

import { Styled } from './ordered-list.styles';

export type OrderedListProps = {
  noMargin?: boolean;
};

export const OrderedList: React.FC<OrderedListProps> = ({
  children,
  ...props
}) => (
  <Styled.OrderedListContainer {...props}>
    {children}
  </Styled.OrderedListContainer>
);
