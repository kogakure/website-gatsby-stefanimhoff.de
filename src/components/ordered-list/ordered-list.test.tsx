import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../utils/test-utils';

import { Default } from './ordered-list.stories';

describe('OrderedList', () => {
  test('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('is accessible', async () => {
    const { container } = render(<Default />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
