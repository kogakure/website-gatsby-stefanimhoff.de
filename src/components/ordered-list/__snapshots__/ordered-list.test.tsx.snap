// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderedList renders correctly 1`] = `
.c0 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
  padding-left: 0;
}

li .c0 {
  padding-left: 2rem;
}

<ol
  class="c0"
>
  <li>
    Early Stoics
    <ol
      class="c0"
    >
      <li>
        Zeno of Citium
      </li>
      <li>
        Chrysippus
      </li>
      <li>
        Chrysippus
      </li>
    </ol>
  </li>
</ol>
`;
